firebase-app/build.gradle.kts

```diff
@@ -44,47 +47,67 @@ android {
 }
 
 kotlin {
-    js {
-        useCommonJs()
-        nodejs()
-        browser()
-    }
+
     android {
-        publishLibraryVariants("release", "debug")
+        publishAllLibraryVariants()
     }
 
-    val iosArm64 = iosArm64()
-    val iosX64 = iosX64("ios") {
+    fun nativeTargetConfig(): KotlinNativeTarget.() -> Unit = {
+        val nativeFrameworkPaths = listOf(
+            projectDir.resolve("src/nativeInterop/cinterop/Carthage/Build/iOS")
+        )
+
         binaries {
             getTest("DEBUG").apply {
-                linkerOpts("-F$projectDir/src/iosMain/c_interop/Carthage/Build/iOS/")
+                linkerOpts(nativeFrameworkPaths.map { "-F$it" })
                 linkerOpts("-ObjC")
             }
         }
+
+        compilations.getByName("main") {
+            cinterops.create("FirebaseCore") {
+                compilerOpts(nativeFrameworkPaths.map { "-F$it" })
+                extraOpts("-verbose")
+            }
+        }
+    }
+
+    if (project.extra["ideaActive"] as Boolean) {
+        iosX64("ios", nativeTargetConfig())
+    } else {
+        ios(configure = nativeTargetConfig())
+    }
+
+    js {
+        useCommonJs()
+        nodejs()
+        browser()
     }
 
     sourceSets {
+        all {
+            languageSettings.apply {
+                apiVersion = "1.4"
+                languageVersion = "1.4"
+                progressiveMode = true
+            }
+        }
+
         val commonMain by getting {
             dependencies {
                 implementation(project(":firebase-common"))
             }
         }
+
         val androidMain by getting {
             dependencies {
-                api("com.google.firebase:firebase-common")
+                api("com.google.firebase:firebase-common:19.5.0")
```

Comment:
please extract a project variable, I see a different version (20.0.3) in other [place](https://github.com/splendo/firebase-kotlin-sdk/pull/25/files#diff-ce7627173baaf1e1be32f69168950f18198b95b959f9ca702440cb060c45db47R131)